src/core/generator/graphql_type.rs

```diff
@@ -1,122 +1,116 @@
 use std::fmt::Display;
 
 use convert_case::{Case, Casing};
-pub(super) static DEFAULT_SEPARATOR: &str = "_";
+pub(super) static DEFAULT_SEPARATOR: &str = "__";
 static PACKAGE_SEPARATOR: &str = ".";
 
+fn normalize_name(name: &str) -> String {
+    name.replace(PACKAGE_SEPARATOR, "_")
+}
+
 /// A struct to represent the name of a GraphQL type.
 #[derive(Debug, Clone)]
 pub struct GraphQLType<A>(A);
 
 #[derive(Debug, Clone)]
 pub struct Parsed {
-    package: Option<Package>,
+    namespace: Option<Namespace>,
     name: String,
     entity: Entity,
 }
 
 #[derive(Debug, Clone)]
 pub struct Unparsed {
-    package: Option<String>,
+    namespace: Option<Namespace>,
     name: String,
 }
 
 #[derive(Debug, Clone)]
-struct Package {
+struct Namespace {
     path: Vec<String>,
-    input: String,
 }
 
-impl Package {
-    fn parse(input: &str) -> Option<Self> {
-        let separator = PACKAGE_SEPARATOR;
-        let path = input.split(separator).map(String::from).collect::<Vec<_>>();
-        if path.is_empty() | input.is_empty() {
-            None
-        } else {
-            Some(Self { path, input: input.to_string() })
-        }
+impl Namespace {
+    fn parse(input: &str) -> Self {
+        let path = input
+            .split(PACKAGE_SEPARATOR)
+            .map(String::from)
+            .collect::<Vec<_>>();
+
+        Self { path }
+    }
+
+    fn combine(mut self, other: Namespace) -> Self {
+        self.path.extend(other.path);
+
+        self
     }
 
-    fn source(&self) -> &str {
-        &self.input
+    fn id(&self) -> String {
+        self.path.join(PACKAGE_SEPARATOR)
```

Comment:
`source` was providing the original name of the type.  What is it doing now?

Reply:
I removed `source`. The id for methods is generated based on added namespaces